src/msix/common/Encoding.cpp

```diff
@@ -301,11 +301,14 @@ namespace MSIX { namespace Encoding {
                                 else { ThrowErrorAndLog(Error::UnknownFileNameEncoding, "Unexpected next sequence value"); }
                             }
 
-                            if (!done)
-                            {   // We are not done! Point to the next % encoded UTF-8 seq
-                                sequenceIndex++;
-                                index++;
+                            if (done)
+                            {   
+                                break;
                             }

```

Comment:
If we are going to "optimize" it, it can currently just be a removal of done and a break where it was set to true.  If this was business logic I would worry about the potential for something to get skipped in the future, but it is unlikely to be the case here.